tests/acceptance/test_opensafely_talk.py

```diff
@@ -0,0 +1,77 @@
+from databuilder import codelist, tables, variables
+from databuilder.query_language import Dataset
+
+
+def test_age_reuse():
+    dataset = Dataset()
+    dataset.age = variables.age(at="2022-01-01")
+
+
+def test_age_raw():
+    patients = tables.patients
+
+    dataset = Dataset()
+    dob = patients.date_of_birth
+    dataset.age = ("2022-01-01" - dob).convert_to_years()
+
+
+def test_has_died_reuse():
+    dataset = Dataset()
+    dataset.age = variables.has_died(at="2022-01-01")
+
+
+def test_has_died_raw():
+    index_date = "2022-01-01"
+    deaths = tables.ons_deaths
+
+    dataset = Dataset()
+    dataset.has_died = deaths                     \
+        .take(deaths.date_of_death <= index_date) \
+        .exists_for_patient()
+
+
+def test_bmi_reuse():
+    dataset = Dataset()
+    dataset.bmi = variables.bmi(
+        index_date="2022-01-01", earliest_date="2012-01-01", min_age=16
+    )
+
+
+def test_bmi_raw():
+    height_codelist = codelist(["XM01E", "229.."], "ctv3")
+    weight_codelist = codelist(["X76C7", "22A.."], "ctv3")
+    bmi_codelist = codelist(["22K.."], "ctv3")
+
+    index_date = "2022-01-01"
+    earliest_date = "2012-01-01"
+    min_age = 16
+
+    age = variables.age
+    events = tables.events
+
+    dataset = Dataset()
+
+    height = events                               \
+        .take(events.code.is_in(height_codelist)) \
+        .take(events.date <= index_date)          \
+        .take(age(at=events.date) >= min_age)     \
+        .sort_by(events.date).last_for_patient()  \
+        .value
```

Comment:
The Black idiom is to use parens to avoid escaping the line-endings e.g.
```py
height = (
         events
        .take(events.code.is_in(height_codelist))
        .take(events.date <= index_date)
        .take(age(at=events.date) >= min_age)
        .sort_by(events.date).last_for_patient()
        .value
)
```
But I agree it's just a choice between one form of distracting syntactic noise and another.

Reply:
Yeah. I can just about forgive Black for that normally, but I really don't like it for a DSL.